[0.436s] Invoking command in '/home/vincent/gz/ws_inspect/src/inspection_mission': AMENT_PREFIX_PATH=/home/vincent/gz/ws_inspect/install/inspection_mission:${AMENT_PREFIX_PATH} CONDA_PROMPT_MODIFIER=(ros2) PATH=/home/vincent/gz/ws_inspect/install/inspection_mission/bin:${PATH} PYTEST_ADDOPTS=--tb=short --junit-xml=/home/vincent/gz/ws_inspect/build/inspection_mission/pytest.xml --junit-prefix=inspection_mission -o cache_dir=/home/vincent/gz/ws_inspect/build/inspection_mission/.pytest_cache PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/vincent/gz/ws_inspect/build/inspection_mission/src:/home/vincent/gz/ws_inspect/install/inspection_mission/lib/python3.12/site-packages:${PYTHONPATH} /usr/bin/python3 -m pytest
[1.048s] ============================= test session starts ==============================
[1.048s] platform linux -- Python 3.12.3, pytest-7.4.4, pluggy-1.4.0
[1.048s] cachedir: /home/vincent/gz/ws_inspect/build/inspection_mission/.pytest_cache
[1.048s] rootdir: /home/vincent/gz/ws_inspect/src/inspection_mission
[1.048s] plugins: launch-testing-ros-0.26.9, ament-pep257-0.17.3, ament-lint-0.17.3, launch-testing-3.4.6, ament-copyright-0.17.3, ament-xmllint-0.17.3, ament-flake8-0.17.3, colcon-core-0.20.0, cov-4.1.0
[1.049s] collected 2 items
[1.049s] 
[1.116s] =============================== warnings summary ===============================
[1.116s] tests/test_mission_node.py::test_waypoint_sequence_advances_and_completes
[1.116s] tests/test_mission_node.py::test_alarm_triggers_cancel_when_requested
[1.116s]   Warning: Failed to fini rosout publisher: rcl node implementation is invalid, at ./src/rcl/node.c:390
[1.116s] 
[1.116s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[1.120s] tests/test_mission_node.py FF                                            [100%]
[1.120s] 
[1.120s] =================================== FAILURES ===================================
[1.120s] ________________ test_waypoint_sequence_advances_and_completes _________________
[1.120s] tests/test_mission_node.py:80: in test_waypoint_sequence_advances_and_completes
[1.120s]     node = MissionNode(action_client_factory=lambda node: mock_client)
[1.120s] src/inspection_mission/mission_node.py:37: in __init__
[1.120s]     super().__init__("mission_node")
[1.120s] /opt/ros/jazzy/lib/python3.12/site-packages/rclpy/node.py:187: in __init__
[1.120s]     self.__node = _rclpy.Node(
[1.120s] E   rclpy._rclpy_pybind11.RCLError: error creating node: __create_participant failed to create participant, at ./src/participant.cpp:107, at ./src/rcl/node.c:252
[1.120s] ----------------------------- Captured stdout call -----------------------------
[1.120s] [37;1m2025-10-03 01:00:49.438 [31;1m[[37;1mRTPS_TRANSPORT_SHM[31;1m Error] [37mFailed to create segment 2e9a401eb9bdc360: Permission denied[34;1m -> Function [36mcompute_per_allocation_extra_size[m
[1.120s] [37;1m2025-10-03 01:00:49.439 [31;1m[[37;1mRTPS_TRANSPORT_SHM[31;1m Error] [37mFailed to create Shared Memory Manager for domain fastrtps: Permission denied[34;1m -> Function [36mcreate[m
[1.120s] [37;1m2025-10-03 01:00:49.439 [31;1m[[37;1mRTPS_PARTICIPANT[31;1m Error] [37mUnable to Register SHM Transport. SHM Transport is not supported in the current platform.[34;1m -> Function [36mRTPSParticipantImpl[m
[1.120s] [37;1m2025-10-03 01:00:49.439 [31;1m[[37;1mTRANSPORT_UDP[31;1m Error] [37mError creating socket: Operation not permitted[34;1m -> Function [36minit[m
[1.120s] [37;1m2025-10-03 01:00:49.439 [31;1m[[37;1mRTPS_PARTICIPANT[31;1m Error] [37mUser transport failed to register.[34;1m -> Function [36mRTPSParticipantImpl[m
[1.120s] [37;1m2025-10-03 01:00:49.439 [31;1m[[37;1mRTPS_PARTICIPANT[31;1m Error] [37mCannot create participant due to initialization error[34;1m -> Function [36mcreateParticipant[m
[1.120s] [37;1m2025-10-03 01:00:49.439 [31;1m[[37;1mDOMAIN_PARTICIPANT[31;1m Error] [37mProblem creating RTPSParticipant[34;1m -> Function [36menable[m
[1.120s] ----------------------------- Captured stderr call -----------------------------
[1.120s] getifaddrs: Operation not permitted
[1.120s] getifaddrs: Operation not permitted
[1.121s] getifaddrs: Operation not permitted
[1.121s] __________________ test_alarm_triggers_cancel_when_requested ___________________
[1.121s] tests/test_mission_node.py:100: in test_alarm_triggers_cancel_when_requested
[1.121s]     node = MissionNode(action_client_factory=lambda node: mock_client)
[1.121s] src/inspection_mission/mission_node.py:37: in __init__
[1.121s]     super().__init__("mission_node")
[1.121s] /opt/ros/jazzy/lib/python3.12/site-packages/rclpy/node.py:187: in __init__
[1.121s]     self.__node = _rclpy.Node(
[1.121s] E   rclpy._rclpy_pybind11.RCLError: error creating node: __create_participant failed to create participant, at ./src/participant.cpp:107, at ./src/rcl/node.c:252
[1.121s] ----------------------------- Captured stdout call -----------------------------
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mRTPS_TRANSPORT_SHM[31;1m Error] [37mFailed to create segment 97d1bb1630dcb2c8: Permission denied[34;1m -> Function [36mcompute_per_allocation_extra_size[m
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mRTPS_TRANSPORT_SHM[31;1m Error] [37mFailed to create Shared Memory Manager for domain fastrtps: Permission denied[34;1m -> Function [36mcreate[m
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mRTPS_PARTICIPANT[31;1m Error] [37mUnable to Register SHM Transport. SHM Transport is not supported in the current platform.[34;1m -> Function [36mRTPSParticipantImpl[m
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mTRANSPORT_UDP[31;1m Error] [37mError creating socket: Operation not permitted[34;1m -> Function [36minit[m
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mRTPS_PARTICIPANT[31;1m Error] [37mUser transport failed to register.[34;1m -> Function [36mRTPSParticipantImpl[m
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mRTPS_PARTICIPANT[31;1m Error] [37mCannot create participant due to initialization error[34;1m -> Function [36mcreateParticipant[m
[1.121s] [37;1m2025-10-03 01:00:49.487 [31;1m[[37;1mDOMAIN_PARTICIPANT[31;1m Error] [37mProblem creating RTPSParticipant[34;1m -> Function [36menable[m
[1.121s] - generated xml file: /home/vincent/gz/ws_inspect/build/inspection_mission/pytest.xml -
[1.121s] =========================== short test summary info ============================
[1.121s] FAILED tests/test_mission_node.py::test_waypoint_sequence_advances_and_completes
[1.121s] FAILED tests/test_mission_node.py::test_alarm_triggers_cancel_when_requested
[1.121s] ======================== 2 failed, 2 warnings in 0.39s =========================
[1.271s] Invoked command in '/home/vincent/gz/ws_inspect/src/inspection_mission' returned '1': AMENT_PREFIX_PATH=/home/vincent/gz/ws_inspect/install/inspection_mission:${AMENT_PREFIX_PATH} CONDA_PROMPT_MODIFIER=(ros2) PATH=/home/vincent/gz/ws_inspect/install/inspection_mission/bin:${PATH} PYTEST_ADDOPTS=--tb=short --junit-xml=/home/vincent/gz/ws_inspect/build/inspection_mission/pytest.xml --junit-prefix=inspection_mission -o cache_dir=/home/vincent/gz/ws_inspect/build/inspection_mission/.pytest_cache PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/vincent/gz/ws_inspect/build/inspection_mission/src:/home/vincent/gz/ws_inspect/install/inspection_mission/lib/python3.12/site-packages:${PYTHONPATH} /usr/bin/python3 -m pytest
